src/parser.h

```diff
@@ -239,6 +239,12 @@ struct eval_res_t {
         : status(status), break_expand(break_expand), was_empty(was_empty), no_status(no_status) {}
 };
 
+enum class parser_status_var_t : uint8_t {
```

Comment:
Unless we expect to have a lot more of these variables, I think we can do without this enum. In every call site we know which status variable we're accessing so we don't benefit from having these be indexable.

Also (as your changes illustrate) any time we want to set `current_command` we probably also want to set `current_commandline`. We can structure these so we're less likely to forget.

I suggest just making a struct:

    struct status_vars_t {
        wcstring command;
        wcstring commandline;
    };
    
and then you can set and get that struct. We don't even really need a setter or getter, just put it in `library_data_t`.

other than that this looks good to me